File: src/main/java/ru/urfu/cake/shop/identity/service/model/UserModel.java
package ru.urfu.cake.shop.identity.service.model;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.persistence.Embedded;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.Set;
import java.util.UUID;

@Data
@Builder
@Schema(description = "Модель пользователя")
public class UserModel {

    @Schema(description = "Уникальный идентификатор пользователя")
    private UUID id;

    @Schema(description = "Электронная почта пользователя")
    private String email;

    @Schema(description = "Телефон пользователя")
    private String phoneNumber;

    @Schema(description = "Имя пользователя")
    private String firstName;

    @Schema(description = "Фамилия пользователя")
    private String lastName;

    @Schema(description = "Отчество пользователя")
    private String middleName;

    @Schema(description = "Адрес пользователя")
    @Embedded
    private AddressModel address;

    @Schema(description = "Ест ли у пользователь сахар")
    private Boolean hasSugar;

    @Schema(description = "Дата и время создания пользователя")
    private LocalDateTime createdAt;

    @Schema(description = "Дата и время последнего обновления пользователя")
    private LocalDateTime updatedAt;

    @Schema(description = "Дата и время последнего входа пользователя")
    private LocalDateTime lastLogin;

    @Schema(description = "Идентификатор корзины пользователя")
    private UUID cartId;

    @Schema(description = "Идентификатор аватара пользователя")
    private UUID avatarImageId;

    @Schema(description = "Дополнительные изображения пользователя")
    private Set<Long> imageIds;

    @Schema(description = "Роли пользователя")
    private Set<String> roles;

    @Schema(description = "Описание, которое может редактировать сам пользователь")
    private String publicDescription;

    @Schema(description = "Внутреннее описание пользователя (видимое только поварам/админу)")
    private String internalDescription;
}

File: src/main/java/ru/urfu/cake/shop/identity/service/model/UserModel.kt
package ru.urfu.cake.shop.identity.service.model

import com.fasterxml.jackson.annotation.JsonInclude
import io.swagger.v3.oas.annotations.media.Schema
import jakarta.persistence.Embedded
import java.time.LocalDateTime
import java.util.UUID

@JsonInclude(JsonInclude.Include.NON_NULL)
@Schema(description = "Модель пользователя")
data class UserModel(
    @field:Schema(description = "Уникальный идентификатор пользователя")
    val id: UUID? = null,

    @field:Schema(description = "Электронная почта пользователя")
    val email: String? = null,

    @field:Schema(description = "Телефон пользователя")
    val phoneNumber: String? = null,

    @field:Schema(description = "Имя пользователя")
    val firstName: String? = null,

    @field:Schema(description = "Фамилия пользователя")
    val lastName: String? = null,

    @field:Schema(description = "Отчество пользователя")
    val middleName: String? = null,

    @field:Schema(description = "Адрес пользователя")
    @Embedded
    val address: AddressModel? = null,

    @field:Schema(description = "Ест ли у пользователь сахар")
    val hasSugar: Boolean? = null,

    @field:Schema(description = "Дата и время создания пользователя")
    val createdAt: LocalDateTime? = null,

    @field:Schema(description = "Дата и время последнего обновления пользователя")
    val updatedAt: LocalDateTime? = null,

    @field:Schema(description = "Дата и время последнего входа пользователя")
    val lastLogin: LocalDateTime? = null,

    @field:Schema(description = "Идентификатор корзины пользователя")
    val cartId: UUID? = null,

    @field:Schema(description = "Идентификатор аватара пользователя")
    val avatarImageId: UUID? = null,

    @field:Schema(description = "Дополнительные изображения пользователя")
    val imageIds: Set<Long>? = null,

    @field:Schema(description = "Роли пользователя")
    val roles: Set<String>? = null,

    @field:Schema(description = "Описание, которое может редактировать сам пользователь")
    val publicDescription: String? = null,

    @field:Schema(description = "Внутреннее описание пользователя (видимое только поварам/админу)")
    val internalDescription: String? = null
)